File: src/main/java/com/my/admin/app/commerce/service/validator/cash/CashStatusValidator.java
package com.my.admin.app.commerce.service.validator.cash;

import com.black.food.manager.exception.CashException;
import com.black.food.manager.exception.ErrorCode;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Component;

import static com.black.food.manager.model.constant.BusinessModelConstant.*;

@Component
public class CashStatusValidator {

    private static final Logger LOGGER = LoggerFactory.getLogger(CashStatusValidator.class);

    public void validateCashStatusRequest(final String statusId) {
        if (statusId.equalsIgnoreCase(CASH_STATUS_INACTIVE_LONG.toString())) {
            LOGGER.error(ErrorCode.INVALID_CASH_STATUS.getDescription());
            throw  new CashException(ErrorCode.INVALID_CASH_STATUS.getDescription(), HttpStatus.CONFLICT);
        }
    }
}

File: src/main/kotlin/com/bk/reactive/app/commerce/admin/my/service/validator/cash/CashStatusValidator.kt
package com.bk.reactive.app.commerce.admin.my.service.validator.cash

import com.bk.reactive.app.commerce.admin.my.exception.api.CashException
import com.bk.reactive.app.commerce.admin.my.exception.code.ErrorCode
import com.bk.reactive.app.commerce.admin.my.model.constant.BusinessModelConstant.Companion.CASH_STATUS_INACTIVE_LONG
import mu.KotlinLogging
import org.springframework.http.HttpStatus
import org.springframework.stereotype.Component
import reactor.core.publisher.Mono

@Component
class CashStatusValidator {
    private val logger = KotlinLogging.logger {}

    fun validateCashStatusRequest(statusId: String): Mono<String> {
        return Mono.just(statusId)
            .map {
                if (statusId == CASH_STATUS_INACTIVE_LONG.toString()) {
                    val message = ErrorCode.INVALID_CASH_STATUS.description

                    logger.error { message }
                    Mono.error<String> {
                        CashException(
                            message = message,
                            code = HttpStatus.CONFLICT,
                            cause = Throwable(message)
                        )
                    }
                }
                it
            }
    }
}